src/Tests/dotnet.Tests/CommandTests/ToolInstallCommandTests.cs

```diff
@@ -28,6 +36,33 @@ public void WhenRunWithBothGlobalAndToolPathShowErrorMessage()
                     "--global --tool-path"));
         }
 
+        [Fact]
+        public void WhenRunWithRoot()
+        {
+            Directory.CreateDirectory("/tmp/folder/sub");
+            var directory = Directory.GetCurrentDirectory();
+            var ridGraphPath = TestContext.GetRuntimeGraphFilePath();
+            try
+            {
+                Directory.SetCurrentDirectory("/tmp/folder");
+
+                new DotnetNewCommand(Log, "tool-manifest").WithCustomHive("/tmp/folder").WithWorkingDirectory("/tmp/folder").Execute().Should().Pass();
+                var parseResult = Parser.Instance.Parse("tool install dotnetsay");
+                new ToolInstallLocalCommand(parseResult, runtimeJsonPathForTests: ridGraphPath).Execute().Should().Be(0);
+
+                Directory.SetCurrentDirectory("/tmp/folder/sub");
```

Comment:
Thank you, that was well written. I am not sure if that's the desired behavior or not. My guess is that this wouldn't work in 8.0.100 if you did not set `isRoot`, because it probably shouldn't replicate the install if `isRoot` is not set to true? In this case, I think the test is testing that this works regardless of whether or not `isRoot` is set, but I would expect it to only work if it was set even before the 8.0.200 change. I could be wrong!

Reply:
Does 8.0.100 default isRoot to false? If so, then you're correct: it wouldn't (and shouldn't) replicate the install, and this test wouldn't be too useful. I wasn't aware that the default changed so recently, but that's good to keep in mind. I don't think this is likely to be backported to 8.0.100, but if it is, we should definitely make sure to update the test to explicitly set isRoot to true partway through. Thanks for noticing that!

Reply:
I'm not sure about the 8.0.100 behavior. Based on what https://github.com/dotnet/sdk/issues/40655 @havranek1024 said it sounds like it defaults to false. May you try it out please and see?

Reply:
I just tried out 8.0.107-dev (i.e., the current tip of the release/8.0.1xx branch), and it defaulted isRoot to true.

Reply:
Thanks for trying that out, apologies for my delay.